packages/graphql-server/src/resolvers/appointment/api.ts

```diff
@@ -0,0 +1,115 @@
+  try {
+    logger.info('callGetAppointmentsAPI', { traceId, args })
+
+    const createResponse = await fetcher({
+      url: `${URLS.appointments}/?${setQueryParams(args)}`,
+      api: process.env['PLATFORM_API_BASE_URL'],
+      method: 'GET',
+      headers: {
+        authorization: context.authorization,
+        'Content-Type': 'application/json',
+        'api-version': API_VERSION,
+      },
+      body: {},
+    })
+    return createResponse as GetAppointmentsResult
+  } catch (error) {
+    logger.error('callGetAppointmentsAPI', { traceId, error })
+    return errors.generateUserInputError(traceId)
+  }
+}
+export const callGetAppointmentByIdAPI = async (args: GetAppointmentByIdArgs, context: ServerContext) => {
+  const traceId = context.traceId
+  logger.info('callGetAppointmentByIdAPI', { args, traceId })
+  try {
+    const response = await fetcher({
+      url: `${URLS.appointments}/${args.id}`,
+      api: process.env['PLATFORM_API_BASE_URL'],
+      method: 'GET',
+      headers: {
+        authorization: context.authorization,
+        'Content-Type': 'application/json',
+        'api-version': API_VERSION,
+      },
+    })
+    return response as Appointment
+  } catch (error) {
+    logger.error('callGetAppointmentByIdAPI', { traceId, error })
+    return errors.generateUserInputError(traceId)
+  }
+}
+
+export const callUpdateAppointmentByAPI = async (args: UpdateAppointmentArgs, context: ServerContext) => {
+  const traceId = context.traceId
+  logger.info('callUpdateAppointmentByAPI', { args, traceId })
+  try {
+    await fetcher({
+      url: `${URLS.appointments}/${args.id}`,
+      api: process.env['PLATFORM_API_BASE_URL'],
+      body: args,
+      method: 'PATCH',
+      headers: {
+        authorization: context.authorization,
+        'Content-Type': 'application/json',
+        'api-version': API_VERSION,
+        'If-Match': args._eTag,
+      },
+    })
+    const response = await fetcher({
+      url: `${URLS.appointments}/${args.id}`,
+      api: process.env['PLATFORM_API_BASE_URL'],
+      method: 'GET',
+      headers: {
+        authorization: context.authorization,
+        'Content-Type': 'application/json',
+        'api-version': API_VERSION,
+      },
+    })
+    return response as Appointment
+  } catch (error) {
+    logger.error('callUpdateAppointmentByAPI', { traceId, error })
+    return errors.generateUserInputError(traceId)
+  }
+}
+
+export const callCreateAppointmentByAPI = async (args: CreateAppointmentArgs, context: ServerContext) => {
+  const traceId = context.traceId
+  logger.info('callCreateAppointmentByAPI', { args, traceId })
+  try {
+    await fetcher({
+      url: `${URLS.appointments}`,
+      api: process.env['PLATFORM_API_BASE_URL'],
+      method: 'POST',
+      headers: {
+        authorization: context.authorization,
+        'Content-Type': 'application/json',
+        'api-version': API_VERSION,
+      },
+      body: args,
+    })
+    return true
```

Comment:
it should be return new appointment instead of true

Reply:
Create appointment API is currently responding HTTP status 201 with no response if executing successfully. There's no data to trace the created entity. Return Boolean is temporary